File: server/src/main/kotlin/br/com/thiagoodev/blog/modules/publication/domain/entities/Talk.kt
package br.com.thiagoodev.blog.modules.publication.domain.entities

import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime
import java.util.UUID

@Entity
@EntityListeners(AuditingEntityListener::class)
@Table(name = "talks")
class Talk(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val uuid: UUID? = null,
    @Column(nullable = false, unique = true)
    val externalId: UUID,
    @Column(nullable = false)
    var title: String,
    @CreatedDate
    @Column(nullable = false, updatable = false)
    var createdAt: LocalDateTime? = null,
    @LastModifiedDate
    @Column(nullable = false)
    var updatedAt: LocalDateTime? = null,
)

File: client/lib/domain/models/talk.dart
final class Talk {
  final String? uuid;
  final String externalId;
  final String title;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const Talk({
    this.uuid,
    required this.externalId,
    required this.title,
    this.createdAt,
    this.updatedAt,
  });

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is Talk && other.externalId == externalId;
  }

  @override
  int get hashCode => externalId.hashCode;
}